### PR TITLE
[1LP][WIP] Reports timelines

### PR DESCRIPTION
**Pull request by t-fischer**
Test the timelines in Cloud Intel. testing was done manually before. 

PRT failed on 5.8 because of https://bugzilla.redhat.com/show_bug.cgi?id=1564454
 {{pytest: -v -q cfme/tests/intelligence/test_reports_with_timelines.py --use-provider=complete}}

### CODE REVIEW

File: cfme/intelligence/intel_timelines.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from navmazing import NavigateToSibling
+
+from cfme.base.login import BaseLoggedInPage
+from widgetastic_manageiq import TimelinesView, ManageIQTree
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep
+from cfme.base import Server
+from widgetastic_patternfly import Accordion
+from widgetastic.widget import View
```
Comment (quarckster) [inline cfme/intelligence/intel_timelines.py]: **REQUIRED**
```python
from navmazing import NavigateToSibling
from widgetastic_manageiq import TimelinesView, ManageIQTree
from widgetastic_patternfly import Accordion
from widgetastic.widget import View

from cfme.base import Server
from cfme.base.login import BaseLoggedInPage
from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep
```

File: cfme/tests/intelligence/test_reports_with_timelines.py [hunk 2]
```diff
@@ -1,16 +1,251 @@
...
+
+
+@pytest.fixture
+def report_policy_event_week():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['policy_event_week']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
```
Comment (quarckster) [inline cfme/tests/intelligence/test_reports_with_timelines.py]: **REQUIRED**
```python
if report.exists:
    report.delete()
```

File: cfme/tests/intelligence/test_reports_with_timelines.py [hunk 3]
```diff
@@ -1,16 +1,251 @@
...
+        for evt in events_list:
+            if hasattr(evt, 'target_name'):
+                if (evt.target_name == new_vm.name and evt.assigned_profiles
+                        in new_vm.assigned_policy_profiles):
+                    found_events.append(evt)
+        logger.info('found events: %r', str(found_events))
+
+        return bool(found_events)
+
+    wait_for(check_timeline,
```
Comment (quarckster) [inline cfme/tests/intelligence/test_reports_with_timelines.py]: **REQUIRED**
It seems a bit pointless to me. Maybe we can remove it from here.

Comment (t-fischer) [reply cfme/tests/intelligence/test_reports_with_timelines.py]: changed as talked offline

File: cfme/tests/intelligence/test_reports_with_timelines.py (hunk 2 above)
Comment (t-fischer) [reply cfme/tests/intelligence/test_reports_with_timelines.py]: done. Should I also check for existance before creating?

File: cfme/intelligence/intel_timelines.py (hunk 1 above)
Comment (t-fischer) [reply cfme/intelligence/intel_timelines.py]: done

File: cfme/tests/intelligence/test_reports_with_timelines.py (hunk 2 above)
Comment (quarckster) [reply cfme/tests/intelligence/test_reports_with_timelines.py]: No, it's no needed.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/intelligence/test_reports_with_timelines.py [hunk 4]
```diff
@@ -1,16 +1,294 @@
...
+                    break
+
+        logger.info('found events: %r', str(found_events))
+        return bool(found_events)
+
+    wait_for(check_timeline,
+             timeout='5m',
+             fail_func=lambda: navigate_to(report_hosts_data_management, 'Timelines'),
+             message='event of {} not found on report {}'.format(new_vm.name,
+                                                                report_hosts_data_management.title))
```
Comment (quarckster) [inline cfme/tests/intelligence/test_reports_with_timelines.py]: **REQUIRED**
Indentation

File: cfme/tests/intelligence/test_reports_with_timelines.py [hunk 5]
```diff
@@ -1,16 +1,297 @@
...
+    policy events appears in the cloud intel report timelines. the report is a copy of one
+    existing in Reports"""
+    logger.info('Will stop VM %r', new_vm.name)
+    wait_for(new_vm.provider.mgmt.stop_vm, [new_vm.name], timeout='7m',
+             message='Stopping VM failed')
+    view = navigate_to(report_policy_event_week, 'Timelines')
+    assert view.is_timeline_displayed, 'Timeline is not displayed.'
+    view.flash.assert_no_error()
+
+    def check_timeline():
```
Comment (izapolsk) [inline cfme/tests/intelligence/test_reports_with_timelines.py]: REQUIRED: you have this procedure in every test. it is almost the same. I think you need to define it once and use in all functions.

File: widgetastic_manageiq/__init__.py [hunk 6]
```diff
@@ -2468,7 +2468,7 @@ def __repr__(self):
 
     def __init__(self, parent, locator=None, logger=None):
         super(TimelinesChart, self).__init__(parent=parent, logger=logger)
-        self.locator = locator or '//div[contains(@class, "timeline-container")]'
+        self.locator = locator or '//div[@id="tl_div"]'
```
Comment (izapolsk) [inline widgetastic_manageiq/__init__.py]: QUESTION: are you sure this won't break tests against 5.8 ?

Comment (t-fischer) [reply widgetastic_manageiq/__init__.py]: I tried it works

File: cfme/tests/intelligence/test_reports_with_timelines.py (hunk 5 above)
Comment (t-fischer) [reply cfme/tests/intelligence/test_reports_with_timelines.py]: fixed

**Comment by mshriver** [comment]
@izapolsk can you take ownership of these changes and the rebase required? I'm not sure if this is something that has been implemented by another contributor in the meantime. 

Otherwise we should close this PR.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit da4614b3ddc59fda1a599f17315358f844a871b2

The *local* fixture **`new_vm`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *control_policy*
    - *test_custom_report_policy_timeline*
    - *test_custom_report_policy_7_days_timeline*
    - *test_custom_report_power_operations_timeline*
    - *test_custom_report_hosts_data_management_timeline*

The *local* fixture **`control_policy`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *test_custom_report_policy_timeline*
    - *test_custom_report_policy_7_days_timeline*

The *local* fixture **`report_policy_event_week`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *test_custom_report_policy_timeline*

The *local* fixture **`report_policy_7_days`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *test_custom_report_policy_7_days_timeline*

The *local* fixture **`report_power_operations`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *test_custom_report_power_operations_timeline*

The *local* fixture **`report_hosts_data_management`** is used in the following files:
- **cfme/tests/intelligence/test_reports_with_timelines.py**
    - *test_custom_report_hosts_data_management_timeline*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by john-dupuy** [comment]
These tests are already automated, https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/intelligence/test_reports_with_timelines.py

**Pull request closed without merging**